lib/unleash/feature_toggle.rb

```diff
@@ -51,33 +57,46 @@ def self.disabled_variant
 
     private
 
-    def resolve_variant(context, toggle_enabled)
+    def resolve_variant(context, toggle_enabled, variants, group_id)
       return Unleash::FeatureToggle.disabled_variant unless toggle_enabled
-      return Unleash::FeatureToggle.disabled_variant if sum_variant_defs_weights <= 0
+      return Unleash::FeatureToggle.disabled_variant if sum_variant_defs_weights(variants) <= 0
 
-      variant_from_override_match(context) || variant_from_weights(context, resolve_stickiness)
+      variant_from_override_match(context, variants) || variant_from_weights(context, resolve_stickiness(variants), variants, group_id)
     end
 
-    def resolve_stickiness
-      self.variant_definitions&.map(&:stickiness)&.compact&.first || "default"
+    def resolve_stickiness(variants)
+      variants&.map(&:stickiness)&.compact&.first || "default"
     end
 
     # only check if it is enabled, do not do metrics
     def am_enabled?(context)
-      result =
-        if self.enabled
-          self.strategies.empty? ||
-            self.strategies.any? do |s|
-              strategy_enabled?(s, context) && strategy_constraint_matches?(s, context)
-            end
-        else
-          false
+      am_enabled(context)[:result]
+    end
+
+    def am_enabled(context)
+      returnable = {
+        result: false,
+        strategy: false,
+        variants: self.variant_definitions
```

Comment:
I think we do not need to return variants separately, because it is in strategy object anyways.

Reply:
it allows ut to not test for strategy having variants in the outside level. This sets it to the feature_toggle variants and overrides variants if available at strategy level.

Reply:
fixed